### DISCUSSION TITLE
How use pyscript on PC if not have internet?

### DISCUSSION THREAD

**Discussion by gitgugleadik** [pyscript/pyscript#1748] [Q&A]
How use pyscript on PC if not have internet? 
How run pyscript locale ? (no internet)

**Comment by ericECmorlaix** [comment]
> How use pyscript on PC if not have internet? How run pyscript locale ? (no internet)

> If you're using your local file system, you'll need a way to view how your application looks in your browser. To make this work you'll need to serve these files, and if you already have Python installed it can be accomplished with the following command run from your terminal and in the same directory as your files:
>`python3 -m http.server`
>Then point your browser at [http://localhost:8000](localhost:8000). Remember to refresh the page (CTRL-R) to see any updates you may have made.

[source](https://docs.pyscript.net/2023.09.1/beginning-pyscript/)

**Reply by philippjfr** [reply]
This will still not work if you don't have internet. This allows serving the file locally but pyscript will still have to be loaded from a CDN and the Python runtime and any required packages will all be fetched from the internet.

**Comment by JeffersGlass** [comment]
In PyScript Classic (i.e. all current releases, but not the upcoming release of 2023.09.1) you can provide the `interpreters` key in py-config to point PyScript to Pyodide at a specific location. If you [download pyodide](https://pyodide.org/en/stable/usage/downloading-and-deploying.html), you can point PyScript to your offline version and run everything offline.

Note that each release of PyScript is intended for use with a specific Pyodide version (0.23.2 in the case of the 2023.05.1 release), so it would be good to match your downloaded version with the version of PyScript you intend to use.

Here's an example:

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <script defer src="https://pyscript.net/releases/2023.05.1/pyscript.js"></script>
    <link rel="stylesheet" href="https://pyscript.net/releases/2023.05.1/pyscript.css">
</head>
<body>
    <!-- Replace the src url with a link to your local version; you'll need to run a local dev server to host it-->
    <py-config>
        [[interpreters]]
        src = "https://cdn.jsdelivr.net/pyodide/v0.22.1/full/pyodide.js"
        name = "pyodide-0.22.1"
        lang = "python"
    </py-config>
    <py-script>
        import pyodide_js
        print(pyodide_js.version)
    </py-script>
    <!--  prints '0.22.1', showing the specified Pyodide was loaded even though the release defaults to 0.23.2 -->
</body>
</html>
```

Currently, no similar functionality exists in the upcoming release - the `interpreters` key for py-config has been removed. This might be a good reason to re-introduce it, though we'd probably want to restructure/simplify it, and maybe implementing some method of at least warning users if they're loading a mismatched version of Pyodide.

**Reply by WebReflection** [reply]
There are a couple of things we could do here:

  * allow a `[pyodide] version = '...'` in the config (and `[micropython]`) or bring back `[interpreters]` in a simplified fashion
  * actually split-bundle the TOML parser as we do already for plugins, so that beside the interpreter once the module is installed locally all files are there (still beside the interpreters, as these require much more than just JS).
